# These are parameters to be used with the built in controllers for the actuators. This YAML file is used as part of the gazebo.launch file in the jetsoncar_sim_description repository

jetsoncar: # namespace
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 1000
  

  # Front wheel drive setup
  # Velocity Controllers ---------------------------------------
  front_right_wheel_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: front_right_wheel_joint
    pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}
  front_left_wheel_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: front_left_wheel_joint
    pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}

  # Rear wheel dummy controllers to let the wheels spin freely
  # Note that the PID parameters are all 0
  rear_right_wheel_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: rear_right_wheel_joint
    pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}
  rear_left_wheel_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: rear_left_wheel_joint
    pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}

  # Position Controllers ---------------------------------------
  front_left_hinge_position_controller:
    joint: front_left_hinge_joint
    type: effort_controllers/JointPositionController
    pid: {p: 5.0, i: 0.0, d: 0.05}
  front_right_hinge_position_controller:
    joint: front_right_hinge_joint
    type: effort_controllers/JointPositionController
    pid: {p: 5.0, i: 0.0, d: 0.05}


